gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
Oh I'm happy to not use braces if that's the coding style.

> Yes, uninhibiting is currently broken, but that bug is currently only in suspend-dbus(). Let's keep it that way, so it's easy to fix properly later.

Sorry I don't understand this, could you expand? Do you mean there's a bug in the DBus suspension? I thought the issue was with `xdg-screensaver` method not resuming?

> My prefered solution would be adding a return value to suspend-dbus()

I started with that actually but then realised we already has the cookie variable telling us if it worked or not and that's already used in all the other DBus methods, so I thought adding a return value to this function would make it a special case (unless we changed the rest of the DBus related code to use a similar approach).